src/main/java/org/apache/mesos/hdfs/executor/AbstractNodeExecutor.java

```diff
@@ -92,6 +97,35 @@ private void deleteFile(File fileToDelete) {
   }
 
   /**
+   * Create Symbolic Link for the HDFS binary.
+   **/
+  private void createSymbolicLink() {
+    log.info("Creating a symbolic link for HDFS binary");
+    try {
+      // Find Hdfs binary in sandbox
+      File sandboxHdfsBinary = new File(System.getProperty("user.dir"));
+      Path sandboxHdfsBinaryPath = Paths.get(sandboxHdfsBinary.getAbsolutePath());
+
+      // Create mesosphere opt dir (parent dir of the symbolic link) if it does not exist
+      File mesosphereOptDir = new File(HDFSConstants.MESOSPHERE_OPT_PATH);
+      if (!mesosphereOptDir.exists()) {
+        mesosphereOptDir.mkdirs();
+      }
+
+      // Delete and recreate directory for symbolic link every time
+      File hdfsBinaryDir = new File(HDFSConstants.HDFS_BINARY_PATH);
+      if (hdfsBinaryDir.exists()) {
+        deleteFile(hdfsBinaryDir);
+      }
+
+      // Create symbolic link
+      Path hdfsLinkDirPath = Paths.get(HDFSConstants.HDFS_BINARY_PATH);
+      Files.createSymbolicLink(hdfsLinkDirPath, sandboxHdfsBinaryPath);
```

Comment:
Symlink works fine with the full hadoop home. This is actually the entire directory with ../libexec/*. This worked successfully for the hack week and also is working correctly with the latest code.